core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -18,280 +18,416 @@
 
 require('services/external-rte-save.service.ts');
 
-angular.module('oppia').controller('RteHelperModalController', [
-  '$q', '$scope', '$timeout', '$uibModalInstance', 'AlertsService',
-  'AssetsBackendApiService', 'ContextService',
-  'ExternalRteSaveService', 'FocusManagerService',
-  'ImageLocalStorageService', 'ImageUploadHelperService',
-  'attrsCustomizationArgsDict', 'customizationArgSpecs',
-  'IMAGE_SAVE_DESTINATION_LOCAL_STORAGE', 'ENTITY_TYPE',
-  function(
-      $q, $scope, $timeout, $uibModalInstance, AlertsService,
-      AssetsBackendApiService, ContextService,
-      ExternalRteSaveService, FocusManagerService,
-      ImageLocalStorageService, ImageUploadHelperService,
-      attrsCustomizationArgsDict, customizationArgSpecs,
-      IMAGE_SAVE_DESTINATION_LOCAL_STORAGE, ENTITY_TYPE) {
-    var extractVideoIdFromVideoUrl = function(videoUrl) {
-      videoUrl = videoUrl.split(/(vi\/|v=|\/v\/|youtu\.be\/|\/embed\/)/);
-      return (
-        (videoUrl[2] !== undefined) ?
-        videoUrl[2].split(/[^0-9a-z_\-]/i)[0] : videoUrl[0]);
-    };
+import { Component, Input, ViewChild } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { AppConstants } from 'app.constants';
+import cloneDeep from 'lodash/cloneDeep';
+import { AlertsService } from 'services/alerts.service';
+import { AssetsBackendApiService } from 'services/assets-backend-api.service';
+import { ContextService } from 'services/context.service';
+import { ExternalRteSaveService } from 'services/external-rte-save.service';
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+import { ImageUploadHelperService } from 'services/image-upload-helper.service';
+import { ServicesConstants } from 'services/services.constants';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { FormBuilder, FormGroup } from '@angular/forms';
 
-    $scope.customizationArgSpecs = customizationArgSpecs;
+const extractVideoIdFromVideoUrl = function(videoUrl) {
```

Comment:
Why is this declared in global scope? Can we not do that?

Reply:
I don't really understand what you mean by "scope of this file". Don't we concatenate our files when building? Also, do we have this pattern in other files (i.e. functions defined at top level)?

In any case, I think it would still be sensible to move this into the component class since this is the only place in the codebase which uses this function.

Reply:
I have moved into the function into the class.